miasm2/core/ctypesmngr.py

```diff
@@ -140,236 +511,186 @@ def ast_eval_int(self, ast):
             raise NotImplementedError("Not implemented!")
         return result
 
-
-    def ast_get_align_size(self, ast):
-        """Evaluates the size/alignment of a C ast object
-
-        @ast: parsed pycparser.c_ast object
-        """
-
-        if isinstance(ast, c_ast.Decl):
-            return self.ast_get_align_size(ast.type)
-        elif isinstance(ast, c_ast.TypeDecl):
-            return self.ast_get_align_size(ast.type)
-        elif isinstance(ast, c_ast.IdentifierType):
-            assert isinstance(ast, c_ast.IdentifierType)
-            names = ast.names
-            names = tuple(names)
-            if not self.is_known_type(names):
-                raise RuntimeError("Unknown type %r" % names)
-            obj = self.get_type(names)
-        elif isinstance(ast, c_ast.ArrayDecl):
-            subobj = self.ast_get_align_size(ast.type)
-            dim = ast.dim
-            value = self.ast_eval_int(dim)
-            obj = ObjCArray(subobj, value)
-        elif isinstance(ast, c_ast.Union):
-            obj = self.ast_gen_union_align_size(ast)
-        elif isinstance(ast, c_ast.Struct):
-            obj = self.ast_gen_struct_align_size(ast)
-        elif isinstance(ast, c_ast.PtrDecl):
-            void_type = self.void_ptr
-            subobj = self.ast_get_align_size(ast.type)
-            obj = ObjCPtr('noname', subobj, void_type.align, void_type.size)
+    def ast_to_typeid_struct(self, ast):
+        """Return the CTypeBase of an Struct ast"""
+        name = self.gen_uniq_name() if ast.name is None else ast.name
+        args = []
+        if ast.decls:
```

Comment:
What if the expected value of `ast.decls` if it is not set?
If it is `None`, it would be better to check against `None`. If it is an empty iterable, it is not necessary.